### PR TITLE
Implement Info Storage using BoltDB

### PR DESCRIPTION
**Pull request by mbarkhau**

### CODE REVIEW

**Comment by seiflotfy** [comment]
yeah I agree martin here... I also think domain makes more sense for some instinctive feeling

**Comment by mbarkhau** [comment]
My only concern is that people think of domain in terms of DNS.

**Comment by seiflotfy** [comment]
i get that, my problem with key is that everyone i described it too thought of it as a key-value thingie when i say key they expect a value.... lets thinks about this more

**Comment by martinpinto** [comment]
My problem is that key is not specific enough. With domain you can actually encompass a wider meaning. A key to me is something you use in a key-value store to retrieve (single) items, whereas a domain is something you can use to retrieve a whole domain of keys. Domain adds one more layer of granularity.

**Comment by martinpinto** [comment]
But yeah, I agree with seiflotfy and with you mbarkhau, let's think about this more thoroughly...

**Comment by seiflotfy** [comment]
did you consider leveldb since its part of go now, which means no external dependency?

**Comment by martinpinto** [comment]
@seiflotfy will this be a problem when using leveldb? 
- Only a single process (possibly multi-threaded) can access a particular database at a time.

**Comment by martinpinto** [comment]
@mbarkhau, @seiflotfy: regarding the naming problem on whether we call it key or domain, why not just call it for what it is, a "data domain"? I think this terminology makes much sense. In this way we don't need to rename our variables, we just should explain it in our README.md. This is also the official terminology by Flajolet et al.

**Comment by seiflotfy** [comment]
I agree but found the word data redundant since all we work with is data. Thus I thought domain makes sense :D


—

**Comment by martinpinto** [comment]
yeah of course, we only need to describe it in our README.md and that's all

File: counters/manager.go [hunk 1]
```diff
@@ -57,9 +57,9 @@ GetDomains ...
 */
 func (m *ManagerStruct) GetDomains() ([]string, error) {
 	// TODO: Remove dummy data and implement proper result
-	values := manager.cache.Keys()
-	domains := make([]string, len(values), len(values))
-	for i, v := range values {
+	cache_keys := manager.cache.Keys()
```
Comment (seiflotfy) [inline counters/manager.go]: not convention
cacheKeys you python lover

File: server/server.go [hunk 2]
```diff
@@ -27,12 +26,16 @@ var counterManager *counters.ManagerStruct
 /*
 Server manages the http connections and communciates with the counters manager
 */
-type Server struct {
+type Server struct{}
+
+type domainsResult struct {
+	Domains []string `json:"domains"`
```
Comment (seiflotfy) [inline server/server.go]: plz change back to result

File: server/server.go [hunk 3]
```diff
@@ -27,12 +26,16 @@ var counterManager *counters.ManagerStruct
...
+	Error   error    `json:"error"`
 }
 
-type result struct {
-	Result interface{} `json:"result"`
-	Error  error       `json:"error"`
+type domainResult struct {
+	Result uint  `json:"result"`
```
Comment (seiflotfy) [inline server/server.go]: interface is more appropriate you nub

**Pull request merged**